src/pages/settings/Subscription/CardSection/utils.ts

```diff
@@ -0,0 +1,25 @@
+import {addDays, addMonths, differenceInMonths, format, isValid} from 'date-fns';
+import CONST from '@src/CONST';
+
+/**
+ * Get the next billing date.
+ *
+ * @param initialDate - The initial billing date in 'yyyy-MM-dd' format.
+ * @returns - The next billing date in 'yyyy-MM-dd' format.
+ */
+function getNextBillingDate(initialDate: string): string {
+    let start = new Date(`${initialDate}T00:00:00`);
+
+    if (!isValid(start)) {
+        start = new Date();
+    }
+
+    const current = new Date(start);
```

Comment:
a little lost here, isn't `current === start` ? Then what's the thought of adding `differenceInMonths` ?

My code I had used `const today = new Date()`.

Reply:
Problem fixed 
I also found one more edge case and fixed it

Reply:
I raised one more comment.

Reply:
I haven't reviewed the function logic closely but just glancing at the expected next billing dates in the test, I wanted to clarify expected behavior... the next billing date should always be the **first of the next month** ([internal ref](https://expensify.slack.com/archives/C03TQ48KC/p1718287987838569))

Reply:
I don't have the access to the link, but does that mean we don't even need the `initialDate`. We should always:
```
     startOfMonth(addMonths(today, 1))
```

?

Reply:
No worries. Better we caught this now.

Reply:
@pasyukevich Can you take care of this change?

Reply:
Updated